apis/run/v1alpha3/clusterbootstrap_types.go

```diff
@@ -4,12 +4,33 @@
 package v1alpha3
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ClusterBootstrapStatus defines the observed state of ClusterBootstrap
 type ClusterBootstrapStatus struct {
 	ResolvedTKR string `json:"resolvedTKR,omitempty"`
+
+	Conditions Conditions `json:"conditions,omitempty"`
+}
+
+type Conditions []Condition
+type ConditionType string
+
+type Condition struct {
+	// Type of condition in CamelCase
+	Type ConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown
+	Status corev1.ConditionStatus `json:"status"`
+
+	// A human readable error message. Will be empty in case no error happened
+	UsefulErrorMessage string `json:"usefulErrorMessage,omitempty"`
```

Comment:
Don't want to interfere, but I would just call this `ErrorMessage`. I hope most of those fields are useful in some way :)

Reply:
Also it might be worth to align a bit more with the upstream CAPI Condition. I'm not 100% sure if we ourselves are aligning to further upstream standards.

We use: https://github.com/kubernetes-sigs/cluster-api/blob/9377ba2a19ea69f716f88689f6a550698d4da4b2/api/v1beta1/condition_types.go#L79-L84

So it might be worth it using Reason/Message instead of ErrorMessage.

Fabrizio knows more about how we got there.

Reply:
cc @fabriziopandini

Reply:
@sbueringer Carvel kapp controller has ```UsefulErrorMessage``` field for ```pkgi``` https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/pkg/apis/kappctrl/v1alpha1/status.go#L21 
:)

Reply:
Changed to fill ```Message``` with ```UsefulErrorMessage```